### ISSUE TITLE
False positive `len-as-condition` in ternary condition

### ISSUE THREAD

**Issue by bfrascher** [PyCQA/pylint#1325]
### Steps to reproduce
Use pylint on the following minimal code snippet:
```python
l = [1, 2, 3]
len(l) if l else 0
```

### Current behavior

Pylint issues the warning `len-as-condition` on the second line of the above code.

### Expected behavior

Since `len` is not used as a condition in the above code I would not expect pylint to issue the `len-as-condition` warning.

### pylint --version output

```bash
pylint 2.0.0,
astroid 1.5.0
Python 3.6.0 (default, Jan 16 2017, 12:12:55)
[GCC 6.3.1 20170109]
```

**Comment by rogalski** [comment]
Thanks for bug report, I'm able to reproduce an issue.

**Comment by rogalski** [comment]
Fixed by ea964a0
